### PR TITLE
Feat/update country selection

### PR DESCRIPTION
**Pull request by CharlesSquirel**

### CODE REVIEW

File: src/components/TextInputWithDropdown/TextInputWithDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    >
+      <TextInput
+        label="Country of residency"
+        placeholder="Start typing location"
+        value={values.country}
+        onChange={handleChange}
+        name="country"
+        excludeDigits
+      />
+      {values.country.length !== 0 && isDropdownActive && (
```
Comment (Sumick) [inline src/components/TextInputWithDropdown/TextInputWithDropdown.tsx]: **issue:** `isDropdownActive` prevents it from working here

File: src/components/TextInputWithDropdown/TextInputWithDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const isTargetInput = (e.target as HTMLElement).tagName === 'INPUT'
+    if (isTargetInput) {
+      setIsDropdownActive(values.country.length !== 0)
+    }
+  }
+
+  return (
+    <div
+      className={styles.container}
+      onClick={(e) => handleCountryInputClick(e)}
```
Comment (Sumick) [inline src/components/TextInputWithDropdown/TextInputWithDropdown.tsx:25]: **issue:** item selection should happen in DropdownCountry, as `onSelect`

File: src/components/TextInputWithDropdown/TextInputWithDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+const TextInputWithDropdown = () => {
+  const { values, handleChange } = useFormikContext<CreateProfileFormValues>()
+
+  const [isDropdownActive, setIsDropdownActive] = useState(false)
+
+  const handleCountryInputClick = (
+    e: React.MouseEvent<HTMLDivElement>,
+  ): void => {
+    const isTargetInput = (e.target as HTMLElement).tagName === 'INPUT'
```
Comment (Sumick) [inline src/components/TextInputWithDropdown/TextInputWithDropdown.tsx:16]: **nitpick:** This is too hacky - we can do it without this function

File: src/components/TextInputWithDropdown/TextInputWithDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      <TextInput
+        label="Country of residency"
+        placeholder="Start typing location"
+        value={values.country}
+        onChange={handleChange}
+        name="country"
+        excludeDigits
+      />
+      {values.country.length !== 0 && isDropdownActive && (
+        <DropdownCountry
```
Comment (Sumick) [inline src/components/TextInputWithDropdown/TextInputWithDropdown.tsx:36]: **nitpick:** Dropdown should close after selection

**Review by Sumick — CHANGES_REQUESTED** [review]
UI looks good, but it does not work correctly yet - we need few improvements, but you are really close! Good job!

File: src/components/TextInputWithDropdown/TextInputWithDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  const handleClickOutside = (e: MouseEvent) => {
+    const dropdownContainer = dropdownRef.current
+    if (dropdownContainer && !dropdownContainer.contains(e.target as Node)) {
+      setIsDropdownActive(false)
+      preventInvalidCountry()
+    }
+  }
+
+  useEffect(() => {
```
Comment (Sumick) [inline src/components/TextInputWithDropdown/TextInputWithDropdown.tsx:45]: **nitpick:** We should create some reusable hook called `useOutsideClick` so its possible to use it somewhere else

**Review by Sumick — APPROVED** [review]
Looks good, great job! Let's add `useOutsideClick` hook before merging

**Review by mmcode9407 — APPROVED** [review]
Nice!

**Review by wawrzonnn — APPROVED** [review]
Great job! a lot of hard work has gone into this

**Pull request merged**